components/ws-manager/pkg/manager/manager.go

```diff
@@ -217,6 +251,29 @@ func (m *Manager) StartWorkspace(ctx context.Context, req *api.StartWorkspaceReq
 	return okResponse, nil
 }
 
+func podRunning(clientset client.Client, podName, namespace string) wait.ConditionWithContextFunc {
+	return func(ctx context.Context) (bool, error) {
+		var pod corev1.Pod
+		err := clientset.Get(ctx, types.NamespacedName{Namespace: namespace, Name: podName}, &pod)
+		if err != nil {
+			return false, nil
+		}
+
+		switch pod.Status.Phase {
+		case corev1.PodFailed, corev1.PodSucceeded:
+			return false, fmt.Errorf("pod ran to completion")
+		case corev1.PodPending:
+			if pod.Status.Reason == "OutOfmemory" || pod.Status.Reason == "OutOfcpu" {
```

Comment:
Maybe it is like this?
https://github.com/kubernetes/kubernetes/blob/a274ec0afa6202a7da53b25ff3fe22df47a1b45b/CHANGELOG/CHANGELOG-1.22.md#workloads-that-saturate-nodes-with-pods-may-see-pods-that-fail-due-to-node-admission
```suggestion
			if pod.Status.Reason == "OutOfMemory" || pod.Status.Reason == "OutOfCpu" {
```